server/aesdsocket.c

```diff
@@ -86,16 +264,14 @@ int mySocketServer(){
   /***************************** SOCKET VARIABLES ****************************/
   struct sockaddr peerAddr;
   socklen_t peerAddrLen = sizeof(peerAddr);
-  int32_t bytesRcvd, noOfBytesinBuffer = 0;
-  uint32_t numBufFragments = 1;
-  int32_t bytesPerPacket;
-  uint32_t bytesToBeWritten = 0;
-  char recPacket[BUFFER_LENGTH];
-  char* recBuffer = NULL;
-  char* trsBuffer = NULL;
-  
+  slist_data_t *tNode = NULL;
+  SLIST_INIT(&head);
+
   /******************** ACCEPT CONNECTION TILL SIGNAL CAUGHT *******************/
   while(!signalCaught){                                                                            // Execute till SIGTERM or SIGINT //
+    if(signalCaught){
+      break;
+    }
```

Comment:
@Vignesh-Vadivel the extra check for `signalCaught` here is probably not needed, it would only catch the signal which happens between line 271 and line 272.